File: Medium/2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.java
class Solution{
    public int[] findThePrefixCommonArray(int[] a, int[] b){
        int n=a.length;
        int m=b.length;

        int[] prefix=new int[n];
        int c=0;
        HashMap<Integer, Integer> mp=new HashMap<>();
        for(int i=0; i<n; i++){
            if(a[i] == b[i]){
                mp.put(a[i], mp.getOrDefault(a[i], 0)+2);
                c++;
            }
            else{
                mp.put(a[i], mp.getOrDefault(a[i], 0)+1);
                mp.put(b[i], mp.getOrDefault(b[i], 0)+1);
                if(mp.get(a[i]) == 2)
                    c++;
                if(mp.get(b[i]) == 2)
                    c++;
            }
            prefix[i]=c;
        }

        return prefix;
    }
}

File: Medium/2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, a: List[int], b: List[int]) -> List[int]:
        n=len(a)
        m=len(b)

        prefix=[0]*n
        c=0
        mp={}
        for i in range(n):
            if a[i] == b[i]:
                mp[a[i]]=mp.get(a[i], 0)+2
                c+=1
            else:
                mp[a[i]]=mp.get(a[i], 0)+1
                mp[b[i]]=mp.get(b[i], 0)+1
                if mp[a[i]] == 2:
                    c+=1
                if mp[b[i]] == 2:
                    c+=1
            prefix[i]=c

        return prefix